src/thread_pool_async.ml

```diff
@@ -2,10 +2,15 @@ open Base
 open Async_kernel
 open Async_unix
 
-type 'state thread = In_thread.Helper_thread.t * 'state
+type 'state worker = {
+  thread : In_thread.Helper_thread.t;
+  state : 'state;
+}
+
+type 'outcome result = 'outcome Deferred.Or_error.t
```

Comment:
Can we find a different type name for it? `result` is already a pretty well-known built-in type so to avoid confusion I would suggest changing the name.

Reply:
I thought it was very fitting because underneath it *is* a `Deferred.Result.t`.